src/accelerate/utils/environment.py

```diff
@@ -61,17 +62,41 @@ def are_libraries_initialized(*library_names: str) -> Dict[str, bool]:
     return [lib_name for lib_name in library_names if lib_name in sys.modules]
 
 
+def get_gpu_info():
+    """
+    Gets GPU count and names using `nvidia-smi` instead of torch to not initialize CUDA.
+
+    Largely based on the `gputil` library.
+    """
+    if platform.system() == "Windows":
+        # If platform is Windows and nvidia-smi can't be found in path
+        # try from systemd rive with default installation path
+        command = spawn.find_executable("nvidia-smi")
+        if command is None:
+            command = "%s\\Program Files\\NVIDIA Corporation\\NVSMI\\nvidia-smi.exe" % os.environ["systemdrive"]
+    else:
+        command = "nvidia-smi"
+    output = subprocess.check_output([command, "-L"], universal_newlines=True)
+    lines = output.split(os.linesep)
+    gpu_count = len(lines) - 1
+    # Get name from first GPU
+    gpu_name = lines[0].split("(")[0].strip()
+    return gpu_count, gpu_name
+
+
 def check_cuda_p2p_ib_support():
     """
     Checks if the devices being used have issues with P2P and IB communications, namely any consumer GPU hardware after
     the 3090.
+
+    Noteably uses `nvidia-smi` instead of torch to not initialize CUDA.
     """
-    if torch.cuda.is_available():
-        # Get the first device/default
-        device_name = torch.cuda.get_device_name()
-        device_count = torch.cuda.device_count()
+    try:
```

Comment:
Should cuda availability still be checked? Sure, the `try ... except` takes care, but do we ever need to run this if cuda is not available?

Reply:
The issue is we can’t check if CUDA is available. If we do, we’ll hit the CUDA was initialized issue because any call to the `torch.cuda` module will init it

Reply:
And thus leading to an error about CUDA being initialized before the notebook launcher could be ran

Reply:
Nice! This is a much, *much* cleaner implementation!